src/request/TopLevelApi.js

```diff
@@ -63,10 +65,23 @@ class TopLevelApi extends RequestParser { // eslint-disable-line no-unused-vars
             await KeyStore.instance.migrateAccountsToKeys();
         }
 
+        const parsedRequest = await this.parseRequest(request);
+
         return new Promise((resolve, reject) => {
             this._resolve = resolve;
             this._reject = reject;
 
+            if (!parsedRequest) { // should already be rejected here with an Errors.InvalidRequestError()
+                // this really should never happen
+                this.reject(new Errors.InvalidRequestError('Request was not successfully parsed'));
+                return;
+            }
+
+            if (!(/** @type {ParsedSimpleRequest} */(parsedRequest).keyInfo)
+                || /** @type {ParsedSimpleRequest} */(parsedRequest).keyInfo.encrypted) {
```

Comment:
What does that mean conceptually?

Reply:
It means the CryptoWorker is only loaded for requests which do not contain a `keyInfo` in their request (like `create` and `import`) and for requests where the key is `encrypted`.

Reply:
But I agree, a comment explaining this in the code would be good.